test/solver/batch_cg_kernels.cpp

```diff
@@ -0,0 +1,240 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "core/solver/batch_cg_kernels.hpp"
+
+
+#include <memory>
+#include <random>
+
+
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/core/base/batch_multi_vector.hpp>
+#include <ginkgo/core/log/batch_logger.hpp>
+#include <ginkgo/core/matrix/batch_dense.hpp>
+#include <ginkgo/core/matrix/batch_ell.hpp>
+#include <ginkgo/core/solver/batch_cg.hpp>
+
+
+#include "core/base/batch_utilities.hpp"
+#include "core/matrix/batch_dense_kernels.hpp"
+#include "core/test/utils.hpp"
+#include "core/test/utils/batch_helpers.hpp"
+#include "test/utils/executor.hpp"
+
+
+class BatchCg : public CommonTestFixture {
```

Comment:
Since these tests don't compare agains the ref exec, I think it would be fine to merge the tests from `reference/test/...` into this file. I guess that would save a couple of lines.